src/libecalc/common/graph.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+from typing import Dict, Generic, List, Protocol, TypeVar
+
+import networkx as nx
+
+NodeID = str
+
+
+class NodeWithID(Protocol):
+    @property
+    def id(self) -> NodeID:
+        ...
+
+
+TNode = TypeVar("TNode", bound=NodeWithID)
+
+
+class Graph(Generic[TNode]):
+    def __init__(self):
+        self.graph = nx.DiGraph()
```

Comment:
I like this implementation, handling the 2 different perspectives of a graph in different implementations" :) Makes it easier to deal with data I think :)